launch/launch/launch_service.py

```diff
@@ -54,12 +55,15 @@
 # This is only really required pre-3.6, see:
 #   https://github.com/ros2/launch/issues/84
 #   https://bugs.python.org/issue23548
-@atexit.register
-def close_loop():
-    global _g_loops_used
-    for loop in _g_loops_used:
-        if not loop.is_closed():
-            loop.close()
+# Don't add this on Windows. It causes:
+#   https://github.com/ros2/demos/issues/354.
+if os.name != 'nt':
+    @atexit.register
```

Comment:
@ivanpauno you're right about not needing this anymore in Python 3.6. @wjwwood @dirk-thomas is there any intention to (even partially) support Python 3.5? If not, we can simply drop this.

Reply:
We are waiting for @nuclearsandwich to resolve this question. At the moment we claim Dashing supports Python 3 (see [REP 2000](https://github.com/ros-infrastructure/rep/blame/7cbc7ed3c822bd1acae60f43b1d7470d7de52863/rep-2000.rst#L382)) but in fact we use 3.6+ syntax in some Python code (see ros2/ros2#735). Depending on which way we go this might not be necessary.

Reply:
I think we can go ahead with this fix, and delete the `atexit` hook in a follow-up PR if we decide not to support Pyhton 3.5 more.

Reply:
I agree that we can continue with this patch as-is and not change anything substantial w.r.t. Python 3.5 support.

Reply:
Just adding a little info - I think 'launch' is already using some stuff from the 'Types' module that isn't part of python 3.5.  Maybe this was changed, but I think the horse has left the barn on 3.5

Reply:
Specifically, [this, I think](https://github.com/ros2/launch/blob/master/launch/launch/actions/opaque_coroutine.py#L20)
Sorry I said 'types' above but I mean t'typing'

```
$ python3 --version
Python 3.5.2
$ python3 -c 'from typing import Coroutine'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'Coroutine'
```